### ISSUE TITLE
theme specific error pages not working

### ISSUE THREAD

**Issue by havvg** [Sylius/Sylius#5865]
Hi there,

I added `app/Resources/themes/my-theme/TwigBundle/views/Exception/error404.html.twig`. However this file is never read or found by `Symfony\Bundle\TwigBundle\Controller\ExceptionController::findTemplate`. Therefore it's not used. 

Placing the file at `app/Resources/TwigBundle/views/Exception/error404.html.twig` works, but that's not on a per-theme basis. Having both files in place does not work either.

**Comment by havvg** [comment]
I just tinkered around and came up with this code. It's not tested yet, but the issue is the `@Twig/Exception/error404.html.twig` notation, which doesn't pass the `assertResourcePathIsValid` and which is using the shorthand notation of `@Twig` instead of `@TwigBundle/Resources/views` invalidating some assumptions of this locator.

```php
<?php

/*
 * This file is part of the Sylius package.
 *
 * (c) Paweł Jędrzejewski
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */

namespace Sylius\Bundle\ThemeBundle\Locator;

use Sylius\Bundle\ThemeBundle\Model\ThemeInterface;
use Symfony\Component\Filesystem\Filesystem;
use Symfony\Component\HttpKernel\KernelInterface;

/**
 * @author Kamil Kokot <kamil.kokot@lakion.com>
 */
final class BundleResourceLocator implements ResourceLocatorInterface
{
    /**
     * @var Filesystem
     */
    private $filesystem;

    /**
     * @var KernelInterface
     */
    private $kernel;

    /**
     * @param Filesystem $filesystem
     * @param KernelInterface $kernel
     */
    public function __construct(Filesystem $filesystem, KernelInterface $kernel)
    {
        $this->filesystem = $filesystem;
        $this->kernel = $kernel;
    }

    /**
     * {@inheritdoc}
     *
     * @param string $resourcePath Eg. "@AcmeBundle/Resources/views/template.html.twig"
     */
    public function locateResource($resourcePath, ThemeInterface $theme)
    {
        $this->assertResourcePathIsValid($resourcePath);

        $bundleName = $this->getBundleNameFromResourcePath($resourcePath);
        $resourceName = $this->getResourceNameFromResourcePath($resourcePath);

        $bundles = $this->kernel->getBundle($bundleName, false);
        foreach ($bundles as $bundle) {
            $path = sprintf('%s/%s/%s', $theme->getPath(), $bundle->getName(), $resourceName);

            if ($this->filesystem->exists($path)) {
                return $path;
            }
        }

        throw new ResourceNotFoundException($resourcePath, $theme);
    }

    /**
     * @param string $resourcePath
     */
    private function assertResourcePathIsValid($resourcePath)
    {
        if ('@' !== substr($resourcePath, 0, 1)) {
            throw new \InvalidArgumentException(sprintf('Bundle resource path (given "%s") should start with an "@".', $resourcePath));
        }

        if (false !== strpos($resourcePath, '..')) {
            throw new \InvalidArgumentException(sprintf('File name "%s" contains invalid characters (..).', $resourcePath));
        }
    }

    /**
     * @param string $resourcePath
     *
     * @return string
     */
    private function getBundleNameFromResourcePath($resourcePath)
    {
        $bundleName = substr($resourcePath, 1, strpos($resourcePath, '/') - 1);
        if (false === strpos($bundleName, 'Bundle')) {
            $bundleName .= 'Bundle';
        }

        return $bundleName;
    }

    /**
     * @param string $resourcePath
     *
     * @return string
     */
    private function getResourceNameFromResourcePath($resourcePath)
    {
        if (false === strpos($resourcePath, 'Resources/')) {
            return 'views/'.substr($resourcePath, strpos($resourcePath, '/') + 1);
        }

        return substr($resourcePath, strpos($resourcePath, 'Resources/') + strlen('Resources/'));
    }
}
```

**Comment by pamil** [comment]
It should be solved by #5582.
